src/Build/Logging/BinaryLogger/Postprocessing/IBinlogReaderErrors.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Build.Logging
+{
+    /// <summary>
+    /// Type of the error that occurred during reading.
+    /// </summary>
+    public enum ReaderErrorType
+    {
+        /// <summary>
+        /// The file format of the binlog is not supported by the current reader.
+        /// Despite the logs should be supported by older readers - there might be certain format updates that prevent
+        ///  such forward compatibility. The binlog file contains the info about the minimum required reader version
+        ///  to detect this case.
+        /// </summary>
+        UnsupportedFileFormat,
```

Comment:
Currently none.
I was originaly considering to add ability to force read even if the reader thinks it cannot support the binlog - but it's not a good idea :-).

The other case I was considering to use this is when the reader encounters it's not fully up to date (it tries to read newer version of binlog), but it can proceed reading (as minimum required reader version declared by the binlog is lower or equal then current reader version). But raising it as an error might be confusing.

tl;dr;: I'm removing this :-)